jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ExtraDirectoriesParameters.java

```diff
@@ -35,34 +37,51 @@
 
   private final Project project;
 
-  private List<Path> paths;
+  private ListProperty<ExtraDirectoryParameters> paths;
+  private ExtraDirectoryParametersSpec spec;
   private Map<String, String> permissions = Collections.emptyMap();
 
   @Inject
   public ExtraDirectoriesParameters(Project project) {
     this.project = project;
-    paths =
-        Collections.singletonList(
-            project.getProjectDir().toPath().resolve("src").resolve("main").resolve("jib"));
+    paths = project.getObjects().listProperty(ExtraDirectoryParameters.class).empty();
+    spec = project.getObjects().newInstance(ExtraDirectoryParametersSpec.class, project, paths);
+  }
+
+  public void paths(Action<? super ExtraDirectoryParametersSpec> action) {
+    action.execute(spec);
   }
 
   @Input
   public List<String> getPathStrings() {
     // Gradle warns about @Input annotations on File objects, so we have to expose a getter for a
     // String to make them go away.
-    return getPaths().stream().map(Path::toString).collect(Collectors.toList());
+    return getPaths()
+        .stream()
+        .map(extraDirectoryParameters -> extraDirectoryParameters.getFrom().toString())
+        .collect(Collectors.toList());
   }
 
   @Internal
-  public List<Path> getPaths() {
+  public List<ExtraDirectoryParameters> getPaths() {
     // Gradle warns about @Input annotations on File objects, so we have to expose a getter for a
     // String to make them go away.
     String property = System.getProperty(PropertyNames.EXTRA_DIRECTORIES_PATHS);
     if (property != null) {
       List<String> pathStrings = ConfigurationPropertyValidator.parseListProperty(property);
-      return pathStrings.stream().map(Paths::get).collect(Collectors.toList());
+      return pathStrings
+          .stream()
+          .map(path -> new ExtraDirectoryParameters(project, Paths.get(path), "/"))
+          .collect(Collectors.toList());
     }
-    return paths;
+    if (paths.get().isEmpty()) {
```

Comment:
The reason I did it this way was because, if this is set in the constructor, calls to `jib.extraDirectories.paths { path {...} }` will end up adding paths along with `src/main/jib` instead of replacing the default.

Reply:
I guess maybe I can reset this in `ExtraDirectoryParametersSpec`.

Reply:
> will end up adding paths along with `src/main/jib`

Aha, good catch.